tests/test_filters.py

```diff
@@ -172,3 +173,12 @@ def test_row_col_medians_with_mask(
     assert isinstance(medians["cols"], np.ndarray)
     np.testing.assert_array_equal(medians["rows"], image_random_row_medians_masked)
     np.testing.assert_array_equal(medians["cols"], image_random_col_medians_masked)
+
+
+def test_non_square_img(test_filters_random: Filters):
+    test_filters_random.images["pixels"]=test_filters_random.images["pixels"][:,0:512]
+    test_filters_random.images["zero_averaged_background"] = test_filters_random.average_background(
+        test_filters_random.images["pixels"], test_filters_random.images["pixels"]
+    )
+    assert isinstance(test_filters_random.images["zero_averaged_background"], np.ndarray)
+    assert test_filters_random.images["zero_averaged_background"].shape==(1024, 512)
```

Comment:
I think it's useful to test that the array we've received is the same as we are expecting and a simple way of doing that is to `.sum()` it up (we could use `np.testing.assert_array_equal()` but then have to have all of the arrays stored somewhere, read them in and compare them).

This led to the above comment about masking the whole image which prevents us from doing so.

Reply:
So my thought process here and for below was that other tests (mainly the mpl-image-compare tests) would check if the functions work as expected and produce the same output as expected. I didn't know if duplicate tests are frowned upon/common practice.

As a result, and to prevent duplicate tests, here we are just testing that the `average_background` function throws no errors when the image is not square, and returns something of an expected type and size - where the contents of the array don't matter due to testing elsewhere.  I should also mention that only the `average_background` is tested as this was the cause of the initial error.

Please let me know if you agree/disagree and I can use `test_filters_random_with_mask` fixture and overwrite the arrays for the "mask" and "pixels" image with `<img>[:,0:512]` and add the sums.

Reply:
Thanks for the explanation @MaxGamill-Sheffield 

Duplicating testing is indeed a bit of a waste of time, but as I tried to convey in my other comment I do think tests should be explicit and check the returned values and not just the properties of the object.

I don't think its much of a change since the default is for `mask=None` and wouldn't slow the tests down dramatically summing an array and testing its value.

```
    test_filters_random.images["zero_averaged_background"] = test_filters_random.average_background(test_filters_random.images["pixels"])
    ...
     assert test_filters_random.images["zero_averaged_background"].sum() == XXX
```

It might even b possible to parameterise the test with the original 1024x1024 `test_filters_random.images["pixels"]` and a chunk of it (although not sure how the fixtures would play with parameterisation as I've never tried.

Reply:
I've tried the fixture with parameterisation before but despite the numpy seed being set, the arrays were different.

I've replaced the `average_background` mask input with `None` and added `assert np.sum(test_filters_random.images["zero_averaged_background"]) == 44426.48188033322` which should resolve these comments